src/main/java/com/powsybl/openloadflow/ac/outerloop/AcEmulationOuterLoop.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2023, Coreso SA (https://www.coreso.eu/) and TSCNET Services GmbH (https://www.tscnet.eu/)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.ac.outerloop;
+
+import com.powsybl.commons.report.ReportNode;
+import com.powsybl.iidm.network.TwoSides;
+import com.powsybl.openloadflow.ac.AcLoadFlowContext;
+import com.powsybl.openloadflow.ac.AcLoadFlowParameters;
+import com.powsybl.openloadflow.ac.AcOuterLoopContext;
+import com.powsybl.openloadflow.ac.equations.AcEquationType;
+import com.powsybl.openloadflow.ac.equations.AcVariableType;
+import com.powsybl.openloadflow.lf.outerloop.AbstractAcEmulationOuterLoop;
+import com.powsybl.openloadflow.lf.outerloop.OuterLoopResult;
+import com.powsybl.openloadflow.lf.outerloop.OuterLoopStatus;
+import com.powsybl.openloadflow.network.LfHvdc;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Hadrien Godard {@literal <hadrien.godard at artelys.com>}
+ */
+public class AcEmulationOuterLoop
+        extends AbstractAcEmulationOuterLoop<AcVariableType, AcEquationType, AcLoadFlowParameters, AcLoadFlowContext, AcOuterLoopContext>
+        implements AcOuterLoop {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AcEmulationOuterLoop.class);
+    public static final String NAME = "AcEmulation";
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    private boolean checkFeedingSide(LfHvdc hvdc, ContextData contextData) {
+        String hvdcId = hvdc.getId();
+        LfHvdc.AcEmulationControl acEmulationControl = hvdc.getAcEmulationControl();
+
+        if (acEmulationControl.getFeedingSide() == TwoSides.ONE) {
+            if (hvdc.getP1().eval() < 0) {
+                // Switch feeding side
+                LOGGER.trace("Switching feeding side from One to Two for Hvdc: {}", hvdcId);
+                contextData.incrementFeedingSideSwitchCount(hvdcId);
+                hvdc.updateFeedingSide(TwoSides.TWO);
+                if (contextData.getFeedingSideSwitchCount(hvdcId) == MAX_FEEDING_SIDE_SWITCH) {
+                    LOGGER.debug("Two many feeding side switches (flow blocked to 0 MW) for Hvdc: {}", hvdcId);
```

Comment:
I have a problem with this NULL status.
In the end it would produce a solution with status FULLY CONVERGED and where the HVDC transports no power.
But it may not represent the reality and be misleading. And cause users of the simulator to make wrong decisions.

For example the flip fop may result from changes of Phase Tap changers, or other automates, that changes the active power routes. It is not necessarily that the active power in the HVDC has small oscillations around 0 and can be neglected.

Have you found examples where small endless oscillations happen here ? Wouldn't a dead angle zone for which P is 0 be better ?

Reply:
The idea here is that, if the HVDC is changing its feeding side multiple times during the LF algorithm, then its flow should be very close to 0. In realistic case, the feeding side should never change at all during iterations.

We do not have any cases where endless oscillations around zero occur, but in theory, without this security, it may happen. It is mainly because of the discontinuity of the derivative from k to -k around 0.